api/v1alpha6/conversion_test.go

```diff
@@ -25,16 +25,32 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	runtime "k8s.io/apimachinery/pkg/runtime"
 	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/utils/pointer"
 	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
-	ctrlconversion "sigs.k8s.io/controller-runtime/pkg/conversion"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
 
 	infrav1 "sigs.k8s.io/cluster-api-provider-openstack/api/v1alpha7"
+	testhelpers "sigs.k8s.io/cluster-api-provider-openstack/test/helpers"
 )
 
+// Setting this to false to avoid running tests in parallel. Only for use in development.
+const parallel = true
+
+func runParallel(f func(t *testing.T)) func(t *testing.T) {
+	if parallel {
+		return func(t *testing.T) {
+			t.Helper()
+			t.Parallel()
+			f(t)
+		}
+	}
+	return f
+}
```

Comment:
It allows all 8 tests defined in TestFuzzyConversion to run in parallel. Note that it's used explicitly for tests which can run in parallel. It saves quite a chunk of time as they're pretty slow (they do 10,000 iterations each) and mostly CPU bound. It's probably not super important in a CI run, because even sequentially they take way less time than an e2e test. I can remove it if you'd like.

FWIW I'm not a fan of Ginkgo. I find it ugly and mysterious. Gomega is great, though.

Reply:
I see (tested it even!) that makes sense. I thought the other packages would anyway take longer to run but they don't.
No need to change it. Ginkgo would solve this elegantly though :wink: